### ISSUE TITLE
Can't find sensor

### ISSUE THREAD

**Issue by julianrinaldi** [custom-components/youtube#16]
What is the sensor that this integration makes named? I tried searching for "youtube" in my entities and didn't find anything. Does this integration still work?

**Comment by barto95100** [comment]
@julianrinaldi 
I had the same reflection, in fact the sensor has the name sensor.NAME_channel_Youtube

example:

sensor:
  platform: youtube
  channel_id: UCRJE6Yb_R3Xei-QGcy_Qwhw

the sensor name is :

sensor.journal_de_thomas

👍

**Comment by julianrinaldi** [comment]
I'm not seeing that sensor when using your information or the information for the channel I want to track. I'm also not seeing any errors.

**Comment by barto95100** [comment]
What's your channel_id ? I test on my HA

**Comment by julianrinaldi** [comment]
UCtI0Hodo5o5dUb67FeUjDeA

The chanell ID you posted doesn't work for me either, so I'm assuming it's a problem with my installation.

**Comment by barto95100** [comment]
OK I test And the id you send me is for youtube channel SpaceX

this is the sensor : 

https://ibb.co/zhg0Lt0

**Comment by pinkywafer** [comment]
Working here also. Your Id gives me a `sensor.spacex`
Check that your indentation is right where you configured the sensor. It should be:
```yaml
sensor:
  - platform: youtube
    channel_id: UCtI0Hodo5o5dUb67FeUjDeA
```

**Comment by julianrinaldi** [comment]
Thanks, the indentation is definitely correct. Not sure what the issue is, especially since I'm not seeing any errors anywhere.

**Comment by julianrinaldi** [comment]
Is there anyway to debug this?

**Comment by pinkywafer** [comment]
ok, first question, how did you install the youtube custom component?

**Comment by julianrinaldi** [comment]
Through HACS. I can confirm that the correct files are in the custom components folder.

**Comment by pinkywafer** [comment]
ok, have you restarted home assistant after adding to the config?

**Comment by julianrinaldi** [comment]
Yes, many times at this point.

**Comment by julianrinaldi** [comment]
For the record, I'm running Home Assistant on MacOS. Not sure if that matters.

**Comment by pinkywafer** [comment]
no errors in log at all? - do you have:
`WARNING (MainThread) [homeassistant.loader] You are using a custom integration for youtube which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.` at the top of the log?

**Comment by julianrinaldi** [comment]
Yes.

I get errors, but none I can nail down to this component. I have a very extensive Home Assistant setup:

`2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for anniversaries which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for mail_and_packages which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for meural which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for browser_mod which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for garbage_collection which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for simpleicons which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for nodered which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for tuya_custom which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for spacex which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for unifiprotect which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for circadian_lighting which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for eventsensor which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for litter_robot which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for sonos_alarm which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for bhyve which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for apple_tv which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for meross_cloud which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for alexa_media which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for hpprinter which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:23 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for blitzortung which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:25 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for average which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:25 ERROR (MainThread) [homeassistant.components.openuv] Error during uv index data update: Error requesting data from uv: 500, message='Internal Server Error', url=URL('CENSORED')
2020-11-19 14:12:27 ERROR (MainThread) [homeassistant.setup] Setup failed for airvisual: unknown error
Traceback (most recent call last):
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/setup.py", line 166, in _async_setup_component
component = integration.get_component()
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/loader.py", line 416, in get_component
cache[self.domain] = importlib.import_module(self.pkg_path)
File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
return _bootstrap._gcd_import(name[level:], package, level)
File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
File "<frozen importlib._bootstrap_external>", line 790, in exec_module
File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/components/airvisual/__init__.py", line 6, in <module>
from pyairvisual import CloudAPI, NodeSamba
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/pyairvisual/__init__.py", line 2, in <module>
from .cloud_api import CloudAPI # noqa
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/pyairvisual/cloud_api.py", line 9, in <module>
from .node import NodeCloudAPI
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/pyairvisual/node.py", line 12, in <module>
import numpy as np
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/numpy/__init__.py", line 286, in <module>
raise RuntimeError(msg)
RuntimeError: Polyfit sanity test emitted a warning, most likely due to using a buggy Accelerate backend. If you compiled yourself, see site.cfg.example for information. Otherwise report this to the vendor that provided NumPy.
RankWarning: Polyfit may be poorly conditioned
2020-11-19 14:12:28 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140207433555104] Error handling message: Unknown error
Traceback (most recent call last):
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/components/websocket_api/connection.py", line 95, in async_handle
handler(self.hass, self, schema(msg))
File "/Users/homeassistant/.homeassistant/custom_components/browser_mod/connection.py", line 42, in handle_update
devices[deviceID].update(msg.get("data", None))
File "/Users/homeassistant/.homeassistant/custom_components/browser_mod/connection.py", line 83, in update
self.sensor = self.sensor or create_entity(
File "/Users/homeassistant/.homeassistant/custom_components/browser_mod/helpers.py", line 47, in create_entity
adder = hass.data[DOMAIN][DATA_ADDERS][platform]
KeyError: 'sensor'
2020-11-19 14:12:30 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for nws_alerts which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:30 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for sun2 which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:30 WARNING (MainThread) [custom_components.meross_cloud]
===============================
Meross Cloud Custom component
Developed by Alberto Geniola
Low level library version: 0.4.0.5
-------------------------------
This custom component is under development and not yet ready for production use.
In case of errors/misbehave, please report it here:
https://github.com/albertogeniola/meross-homeassistant/issues
If you like this extension and you want to support it, please consider donating.
-------------------------------
List of devices reported by HTTP API:
- Garage Door Opener (msg100) - OnlineStatus.ONLINE
===============================
2020-11-19 14:12:32 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for faastatus which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:32 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for climacell which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:32 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for youtube which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:32 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for attributes which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:32 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for google_geocode which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:36 WARNING (MainThread) [homeassistant.components.hue.light] Please check for software updates of the Garage Door Opener Light bulb in the Philips Hue App.
2020-11-19 14:12:36 WARNING (MainThread) [homeassistant.components.onvif] The date/time on the device (UTC) is '2020-11-19 19:26:57+00:00', which is different from the system '2020-11-19 19:12:35.371996+00:00', this could lead to authentication issues
2020-11-19 14:12:36 WARNING (MainThread) [homeassistant.components.onvif] The date/time on the device (UTC) is '2020-11-19 19:33:16+00:00', which is different from the system '2020-11-19 19:12:35.383729+00:00', this could lead to authentication issues
2020-11-19 14:12:36 WARNING (MainThread) [homeassistant.components.onvif] The date/time on the device (UTC) is '2020-11-19 19:24:07+00:00', which is different from the system '2020-11-19 19:12:35.377201+00:00', this could lead to authentication issues
2020-11-19 14:12:36 WARNING (MainThread) [homeassistant.components.onvif] The date/time on the device (UTC) is '2020-11-19 19:20:32+00:00', which is different from the system '2020-11-19 19:12:35.389283+00:00', this could lead to authentication issues
2020-11-19 14:12:36 WARNING (MainThread) [homeassistant.components.onvif] The date/time on the device (UTC) is '2020-11-19 19:20:54+00:00', which is different from the system '2020-11-19 19:12:35.401670+00:00', this could lead to authentication issues
2020-11-19 14:12:36 WARNING (MainThread) [pyatv.mrp.player_state] Received UpdateClientMessage for unknown player com.apple.TVAirPlay
2020-11-19 14:12:36 WARNING (MainThread) [pyatv.mrp.player_state] Received UpdateClientMessage for unknown player com.plexapp.plex
2020-11-19 14:12:36 WARNING (MainThread) [pyatv.mrp.player_state] Received UpdateClientMessage for unknown player com.apple.TVAirPlay
2020-11-19 14:12:36 WARNING (MainThread) [pyatv.mrp.player_state] Received UpdateClientMessage for unknown player com.google.ios.youtube
2020-11-19 14:12:36 WARNING (MainThread) [pyatv.mrp.player_state] Received UpdateClientMessage for unknown player com.apple.TVMusic
2020-11-19 14:12:36 WARNING (MainThread) [pyatv.mrp.player_state] Received UpdateClientMessage for unknown player com.plexapp.plex
2020-11-19 14:12:36 WARNING (MainThread) [pyatv.mrp.player_state] Received UpdateClientMessage for unknown player com.apple.TVAirPlay
2020-11-19 14:12:37 WARNING (MainThread) [pyatv.mrp.player_state] Received UpdateClientMessage for unknown player com.apple.TVAirPlay
2020-11-19 14:12:37 WARNING (MainThread) [pyatv.mrp.player_state] Received UpdateClientMessage for unknown player com.plexapp.plex
2020-11-19 14:12:39 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for hdhomerun which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:41 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for adaptive_lighting which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-11-19 14:12:54 ERROR (Recorder) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/aiopylgtv/webos_client.py", line 373, in consumer_handler
while not closeout_task.done():
UnboundLocalError: local variable 'closeout_task' referenced before assignment
2020-11-19 14:12:54 ERROR (Recorder) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/aiopylgtv/webos_client.py", line 317, in ping_handler
ping_waiter = await ws.ping()
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/websockets/protocol.py", line 734, in ping
await self.ensure_open()
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/websockets/protocol.py", line 803, in ensure_open
raise self.connection_closed_exc()
websockets.exceptions.ConnectionClosedOK: code = 1000 (OK), no reason
2020-11-19 14:13:43 WARNING (MainThread) [homeassistant.components.websocket_api.http.connection] [140205714060144] Disconnected: Did not receive auth message within 10 seconds
2020-11-19 14:15:46 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.philadelphia_intl is taking over 10 seconds
2020-11-19 14:15:46 ERROR (MainThread) [custom_components.faastatus.sensor] [PHL] Could not update the sensor (TimeoutError)
2020-11-19 14:19:59 WARNING (MainThread) [homeassistant.components.media_player] Updating meural media_player took longer than the scheduled update interval 0:00:10
2020-11-19 14:19:59 WARNING (MainThread) [homeassistant.helpers.entity] Update of media_player.meural_picture_frame is taking over 10 seconds
2020-11-19 14:19:59 ERROR (MainThread) [homeassistant.helpers.entity] Update for media_player.meural_picture_frame fails
Traceback (most recent call last):
File "/Users/homeassistant/.homeassistant/custom_components/meural/pymeural.py", line 46, in request
resp = await self.session.request(
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/aiohttp/client.py", line 551, in _request
await resp.start(conn)
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/aiohttp/client_reqrep.py", line 892, in start
message, payload = await self._protocol.read() # type: ignore # noqa
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/aiohttp/streams.py", line 605, in read
await self._waiter
asyncio.exceptions.CancelledError
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/helpers/entity.py", line 278, in async_update_ha_state
await self.async_device_update()
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/helpers/entity.py", line 482, in async_device_update
await task
File "/Users/homeassistant/.homeassistant/custom_components/meural/media_player.py", line 178, in async_update
self._meural_device = await self.meural.get_device(self.meural_device_id)
File "/Users/homeassistant/.homeassistant/custom_components/meural/pymeural.py", line 81, in get_device
return await self.request("get", f"devices/{device_id}")
File "/Users/homeassistant/.homeassistant/custom_components/meural/pymeural.py", line 46, in request
resp = await self.session.request(
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/async_timeout/__init__.py", line 45, in __exit__
self._do_exit(exc_type)
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/async_timeout/__init__.py", line 92, in _do_exit
raise asyncio.TimeoutError
asyncio.exceptions.TimeoutError
2020-11-19 14:24:17 ERROR (MainThread) [homeassistant.core] Error executing service: <ServiceCall lock.unlock (c:d7bcc85bf238d3e763ed32cc53a855b1): entity_id=['lock.front_door_lock', 'lock.garage_door_lock']>
Traceback (most recent call last):
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/core.py", line 1467, in catch_exceptions
await coro_or_task
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/core.py", line 1486, in _execute_service
await handler.job.target(service_call)
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/helpers/entity_component.py", line 204, in handle_service
await self.hass.helpers.service.entity_service_call(
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/helpers/service.py", line 499, in entity_service_call
future.result() # pop exception if have
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/helpers/entity.py", line 664, in async_request_call
await coro
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/helpers/service.py", line 536, in _handle_entity_call
await result
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/components/august/lock.py", line 50, in async_unlock
await self._call_lock_operation(self._data.async_unlock)
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/components/august/lock.py", line 53, in _call_lock_operation
activities = await lock_operation(self._device_id)
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/components/august/__init__.py", line 348, in async_unlock
return await self._async_call_api_op_requires_bridge(
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/homeassistant/components/august/__init__.py", line 361, in _async_call_api_op_requires_bridge
ret = await func(*args, **kwargs)
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/august/api_async.py", line 204, in async_unlock_return_activities
await self._async_unlock(access_token, lock_id)
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/august/api_async.py", line 182, in _async_unlock
return await self._async_call_lock_operation(
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/august/api_async.py", line 148, in _async_call_lock_operation
response = await self._async_dict_to_api(
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/august/api_async.py", line 255, in _async_dict_to_api
_raise_response_exceptions(response)
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/august/api_async.py", line 276, in _raise_response_exceptions
raise err
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/august/api_async.py", line 262, in _raise_response_exceptions
response.raise_for_status()
File "/Users/homeassistant/homeassistant/lib/python3.9/site-packages/aiohttp/client_reqrep.py", line 1002, in raise_for_status
raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 560, message='unknown', url=URL('CENSORED')
 `

**Comment by pinkywafer** [comment]
just looking through your log, the only thing I can see which *may* be causing an issue (though I can't wee why at the moment) is that you are running python 3.9
HA ships normally on python 3.8.
My config is available on github (https://github.com/pinkywafer/Home-Assistant_Config), and I have CI checks using github actions.  One test is to install the same version of homeassistant that I have installed (a python install) and check the config., so I've just created a new branch, changed the CI parameters to run python 3.9 instead of 3.8, and that check now fails.  I'm not sure if this is related to this issue though.

Another thing to try is to enable debug logging for this component:
```yaml
logger:
  default: warning
  logs:
    custom_components.youtube: debug
```
and see if you get anything there
